src/wristpy/processing/metrics.py

```diff
@@ -232,14 +233,15 @@ def _cleanup_isolated_ones_nonwear_value(nonwear_value_array: np.ndarray) -> np.
     Returns:
         The modified nonwear value array.
     """
-    nonwear_value_is_one = np.where(nonwear_value_array == 1)[0]
-
-    for ones_index in nonwear_value_is_one:
-        if ones_index == 0 or ones_index == len(nonwear_value_array) - 1:
-            continue
-        if (nonwear_value_array[ones_index - 1] > 1) and (
-            nonwear_value_array[ones_index + 1] > 1
-        ):
-            nonwear_value_array[ones_index] = 2
+    nonwear_value_array = nonwear_value_array.astype(int)
```

Comment:
Something's going wrong here I think? What is the datatype of nonwear right now? If it's coming in as a boolean then this entire function is pointless (since its only 0's and 1's). If its already an int/float then why cast?

Reply:
Oh right if its a float then `== 1` can fail.

Reply:
It is an np array of floats actually. It goes into polars.sum_horizontal() as an array of bool but after the code snippet below,  it is an array of floats. I was getting a type error ("`TypeError: ufunc 'bitwise_and' not supported for the input types, and the inputs could not be safely coerced to any supported types according to...`") when doing the bitwise_and , prior to casting.

`        max_window_value = np.maximum(
            nonwear_value_array[window_n : window_n + n_short_epoch_in_long_epoch],
            np.repeat(calculated_nonwear_value, n_short_epoch_in_long_epoch),
        )
        
        nonwear_value_array[window_n : window_n + n_short_epoch_in_long_epoch] = (
            max_window_value
        )
`